src/query/api/v1/handler/prom/read.go

```diff
@@ -143,7 +144,7 @@ func (h *readHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 				xhttp.WriteError(w, err)
 			}
 		} else {
-			promErr := res.Err
+			promErr := errs.Cause(res.Err)
```

Comment:
@ryanhall07 do you know if this will/won't affect the casting below?

Reply:
For the context: I needed to get the cause here because otherwise it is unable to match `promql.ErrQueryTimeout` and `promql.ErrQueryCanceled`  since these errors now come wrapped with stack.

Reply:
let's just remove the switch checking altogether and covert it to use the (gross) 1.13 way of checking for an error by casting.

Reply:
Probably I'm missing something but why removing the `switch` will help us here? I'm quite sure current `switch` works correctly because unit tests pass.

Reply:
I agree the switch probably works correctly.

My originally thinking was `pkg/errors` is deprecated and you should be using the std `errors` pkg, which doesn't support `Cause`. However, thinking about it more I'm guessing something in prom 2.26 is using `pkg/errors` to propagate the errors with a stack now. So this probably what we actually want.